File: Backend/app/models/pet_model.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime, LargeBinary
from sqlalchemy.orm import relationship
from datetime import datetime
from app.db.base import Base

class Pet(Base):
    __tablename__ = "pets"

    id = Column(Integer, primary_key=True)
    publisher_id = Column(Integer, ForeignKey("users.id"), nullable=False)

    name = Column(String, nullable=False)
    species = Column(String, nullable=False)
    race = Column(String)
    birth_date = Column(DateTime)
    gender = Column(String)
    description = Column(String)
    image_url = Column(String)
    image_data = Column(LargeBinary)
    status = Column(String)
    modalidad = Column(String, default="sede")  # "sede" o "hogar"
    telefono_contacto = Column(String, nullable=True)

    created_at = Column(DateTime, default=datetime.utcnow)
    deleted_at = Column(DateTime, nullable=True)

    publisher = relationship("User", back_populates="pets")
    adoptions = relationship("Adoption", back_populates="pet")

File: FrontendV2/src/app/shared/models/pet.model.ts
export interface Pet {
  // Campos obligatorios
  id: number;
  publisher_id: number;
  name: string;
  species: string;
  created_at: string;

  // Campos opcionales
  race?: string | null;
  birth_date?: string | null;
  gender?: string | null;
  description?: string | null;
  image_url?: string | null;
  status?:
    | 'AVAILABLE'
    | 'ADOPTED'
    | 'RESERVED'
    | 'UNAVAILABLE'
    | 'PENDING_APPROVAL'
    | 'REJECTED'
    | null;
  modalidad?: 'sede' | 'hogar' | null;
  telefono_contacto?: string | null;
  adopter_name?: string | null;
  adopter_id?: number | null;
  publisher_name?: string | null;
}

export interface PetCreate {
  name: string;
  species: string;
  race?: string | null;
  birth_date?: string | null;
  gender?: string | null;
  description?: string | null;
  image_url?: string | null;
  modalidad?: 'sede' | 'hogar';
  telefono_contacto?: string | null;
}

export interface PetUpdate {
  name?: string | null;
  species?: string | null;
  race?: string | null;
  birth_date?: string | null;
  gender?: string | null;
  description?: string | null;
  image_url?: string | null;
  status?: string | null;
  modalidad?: string | null;
  telefono_contacto?: string | null;
}
